Add "Save Search" button

sarastanton opened the pull request:
This PR adds a button to the Entities Search page which does the following:
- Copies current URL (containing search params) to user's clipboard
- Displays tooltip text on hover to let the user know something will be copied to their clipboard (in case they have something else currently copied that they don't want to lose)
- Disables and displays a message ("Copied") for 1.5 seconds after button click to let the user know something has happened

Related Trello card:
<blockquote class="trello-card"><a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;ekaase8s&#x2F;48-save-search-button">save search button</a></blockquote>

**Visual changes:**
_On hover_
![Screenshot 2023-03-10 at 2 26 18 PM](https://user-images.githubusercontent.com/20568337/224422731-9ef602d9-745f-4bda-985f-0e8b383686f1.png)

_After click_
![Screenshot 2023-03-10 at 2 27 11 PM](https://user-images.githubusercontent.com/20568337/224422787-d7eee233-a64e-46d3-87e2-853603813476.png)

sarastanton commented on src/pages/EntitiesSearchPage/EntitiesSearchPage.jsx:
```diff
@@ -274,6 +294,18 @@ function EntitiesSearch() {
                                 Reset Search
                             </EuiButton>
+
+                            { disableSaveSearch
```
I know it seems odd to conditionally render a tooltip based on button disabled status – this is a workaround to prevent the tooltip from getting stuck on button click, which seems to happen because the button is disabled and then re-enabled after click 😅

blms replied:
Thanks for flagging, makes sense to me.

blms reviewed (changes requested):
Looks great! A few small changes:

- Can you add this save search button to the letters search as well?
- I think users might be confused by "search parameters URL"—maybe just "current search URL"?
- After the above changes, can you run `npx eslint ./src/**/*.js ./src/**/*.jsx --fix` and fix any of the problems/warnings reported by ESLint? Working on a pre-commit hook to do this automatically now!

sarastanton commented:
> * Can you add this save search button to the letters search as well?

Good idea! I extracted the button into its own component to make it easy to add to the letters search (and other places if needed in the future).



> * I think users might be confused by "search parameters URL"—maybe just "current search URL"?

Updated as suggested 👍



> * After the above changes, can you run `npx eslint ./src/**/*.js ./src/**/*.jsx --fix` and fix any of the problems/warnings reported by ESLint? Working on a pre-commit hook to do this automatically now!

Done 👍

sarastanton commented:
@blms  we might need to rethink this – it works locally but not on the live site, and I think that is because the clipboard is only accessible in secure contexts (i.e. https).

blms commented:
Good catch! Let's check in with Jay and see if he has any thoughts; it might be best to serve the dev server over https anyway.

sarastanton commented:
Looks like that fixed it  – all good now!
